backend/controllers/appointments/selectCourtAppointmentRooms.js

```diff
@@ -286,6 +287,29 @@ const selectCourtAppointmentRoomsFactory = ({
         })
       }
 
+      const availableRooms = await availableSlotsService.getAvailableRooms(res.locals, {
+        startTime,
+        endTime,
+        agencyId,
+      })
+
+      const isRoomStillAvailable = selectedRoom => availableRooms.some(room => room.value === Number(selectedRoom))
+
+      const preLocationAvailableOrNotRequired = selectPreAppointmentLocation
```

Comment:
Could this be simplified to `
 const preLocationAvailableOrNotRequired = Boolean(selectPreAppointmentLocation && isRoomStillAvailable(selectPreAppointmentLocation))
  `

Reply:
Not quite the same unfortunately.  Have neatened it up though.